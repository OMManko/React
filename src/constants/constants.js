export const endPoint = 'http://react-cdp-api.herokuapp.com/movies';

export const moviePosterSizes = {
    SMALL: 'moviePoster__small',
    MEDIUM: 'moviePoster__medium'
};

export const sortOptions = {
    RELEASE: 'release',
    RATING: 'vote_average'
};

export const filterOptions = {
    TITLE: 'title',
    GENRE: 'genres'
};


export const actionTypes = {
    SEARCH_BY: 'SEARCH_BY',
    SORT_BY: 'SORT_BY',
    SEARCH_MOVIES: 'SEARCH_MOVIES',
    UPDATE_INPUT_VALUE: 'UPDATE_INPUT_VALUE',
    SET_SELECTED_MOVIE: 'SET_SELECTED_MOVIE',
    RESET_SELECTED_MOVIE: 'RESET_SELECTED_MOVIE'
};

export const initialState = {
    moviesList: [],
    moviesFound: 0,
    searchInputValue: '',
    selectedSortOption: sortOptions.RELEASE,
    selectedFilterOption: filterOptions.TITLE,
    selectedMovieInfo: {}
};
