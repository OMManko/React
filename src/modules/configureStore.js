import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';

import movieAppReducer from '../reducers/reducers';
import { rootSaga } from '.';

const sagaMiddleware = createSagaMiddleware();

export default (initialState) => {
  const store = createStore(movieAppReducer, initialState, applyMiddleware(sagaMiddleware));

  sagaMiddleware.run(rootSaga);
  store.runSaga = () => sagaMiddleware.run(rootSaga);
  store.close = () => store.dispatch(END);

  return store;
};
