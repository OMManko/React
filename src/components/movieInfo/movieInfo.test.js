import { mount } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import MovieInfo from '../movieInfo/movieInfo';
import { initialState } from '../../constants/constants';

const mockStore = configureStore();

let component;
let store;

const mockData = {
  poster_path: 'https://cdn.cnn.com/cnnnext/dam/assets/190315113313-04-mafia-crime-families-have-evolved-file-restricted-use-super-tease.jpg',
  title: 'Pulp Fiction',
  release_date: '1980-05-20',
  genres: ['Adventure', 'Action', 'Science Fiction'],
  runtime: 154,
  overview: 'Some descrition',
};

describe('<MovieInfo />', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    component = mount(<Provider store={store}><MovieInfo movie = {mockData} /></Provider>);
  });

  it('should render movie info and match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
