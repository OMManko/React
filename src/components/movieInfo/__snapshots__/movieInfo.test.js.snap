// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieInfo /> should render movie info and match snapshot 1`] = `undefined`;
